php/php.project/src/org/netbeans/modules/php/project/ComputeTestMethodAnnotations.java

```diff
@@ -0,0 +1,189 @@
+
+    private static final AtomicInteger usagesCount = new AtomicInteger(0);
+
+    private volatile Document handledDocument;
+
+    public static ComputeTestMethodAnnotations getInstance() {
+        return INSTANCE;
+    }
+
+    private ComputeTestMethodAnnotations() {
+    }
+
+    public void register() {
+        if (usagesCount.getAndIncrement() == 0) {
+            EditorRegistry.addPropertyChangeListener(this);
+        }
+    }
+
+    public void unregister() {
+        if (usagesCount.decrementAndGet() == 0) {
+            EditorRegistry.removePropertyChangeListener(this);
+        }
+    }
+
+    @Override
+    public void propertyChange(PropertyChangeEvent event) {
+        JTextComponent textComponent = EditorRegistry.lastFocusedComponent();
+        if (textComponent != null) {
+            Document document = textComponent.getDocument();
+            if (document != null) {
+                FileObject fileObject = NbEditorUtilities.getFileObject(document);
+                if (fileObject != null) {
+                    if (FileUtils.isPhpFile(fileObject)) {
+                        if (event.getPropertyName().equals(EditorRegistry.FOCUS_GAINED_PROPERTY)) {
+                            handleFileChange(document);
+                            document.addDocumentListener(ComputeTestMethodAnnotations.this);
+                        } else if (event.getPropertyName().equals(EditorRegistry.FOCUS_LOST_PROPERTY)) {
+                            document.removeDocumentListener(this);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public void insertUpdate(DocumentEvent event) {
+        handleFileChange(event.getDocument());
+    }
+
+    @Override
+    public void removeUpdate(DocumentEvent event) {
+        handleFileChange(event.getDocument());
+    }
+
+    @Override
+    public void changedUpdate(DocumentEvent event) {
+        handleFileChange(event.getDocument());
+    }
+
+    private void handleFileChange(Document doc) {
+        handledDocument = doc;
+        task.schedule(500);
+    }
+
+    @Override
+    public void run() {
+        List<TestMethod> testMethods = new ArrayList<>();
+
+        List<TestMethod> methods = getTestMethods(handledDocument);
+        testMethods.addAll(methods);
+
+        /*
+         * Apparently, this method should update the list of annotations at each call of this method,
+         * when the passed collection of methods changes.
+         * Вut I didn't manage to achieve correct work in this case.
+         * So I made the method call first with the empty collection, to clear the annotation list,
+         * then already with the method collection.
+         */
+        TestMethodController.setTestMethods(handledDocument, Collections.emptyList());
+        TestMethodController.setTestMethods(handledDocument, testMethods);
```

Comment:
Perhaps better:

```suggestion
        if (!testMethods.isEmpty()) {
            TestMethodController.setTestMethods(handledDocument, testMethods);
        }
```

Right?

Reply:
I've verified that we cannot get the expected behavior if we don't set an empty list...
(A popup list is not shown after we change a method's name.)

![nb-php-gh8142](https://github.com/user-attachments/assets/e5d94c79-3468-435f-bcf7-4ee5552993e9)

Reply:
Interesting.

Reply:
Implementation
https://github.com/apache/netbeans/blob/5df292f3e26837cc2abf1211fde16bfe5d13c45f/ide/gsf.testrunner.ui/src/org/netbeans/modules/gsf/testrunner/ui/api/TestMethodController.java#L48-L84

Reply:
Pinged @jlahoda, maybe he will have time to look into it and verify it.

Reply:
If I understand correctly, he has a different account now. @lahodaj

Reply:
Sorry for the belated reply, I've attempted to fix here:
https://github.com/apache/netbeans/pull/8201